src/mapping/recurrences/grids.jl

```diff
@@ -0,0 +1,169 @@
+#####################################################################################
+# Grid construction
+#####################################################################################
+struct RegularGrid{D, R <: AbstractRange} <: Grid
+    grid_steps::SVector{D, Float64}
+    grid_minima::SVector{D, Float64}
+    grid_maxima::SVector{D, Float64}
+    grid::NTuple{D, R}
+end
+function RegularGrid(grid::NTuple)
+    D = length(grid)
+    grid_steps = SVector{D,Float64}(step.(grid))
+    grid_maxima = SVector{D,Float64}(maximum.(grid))
+    grid_minima = SVector{D,Float64}(minimum.(grid))
+    return RegularGrid(grid_steps, grid_minima, grid_maxima, grid)
+end
+
+minmax_grid_extent(g::RegularGrid) = g.grid_minima, g.grid_maxima
+mean_cell_diagonal(g::RegularGrid{D}) where {D} = norm(g.grid_steps)
+
+struct IrregularGrid{D} <: Grid
+    grid::NTuple{D, Vector{Float64}}
+end
+minmax_grid_extent(g::IrregularGrid) = minmax_grid_extent(g.grid)
+
+minmax_grid_extent(g::NTuple) = map(minimum, g), map(maximum, g)
+function mean_cell_diagonal(g::NTuple)
+    steps = map(r -> mean(diff(r)), g)
+    return norm(steps)
+end
+
+"""
+    SubdivisionBasedGrid(grid::NTuple{D, <:AbstractRange}, lvl_array::Array{Int, D})
+
+Given a coarse `grid` tesselating the state space, construct a `SubdivisionBasedGrid`
+based on the given level array `lvl_array`.
+The level array is has non-negative integer values.
+Value 0 means that the corresponding cell of the coarse `grid` is not subdivided
+any further. Value `n > 0` means that the corresponding cell will be subdivided
+in total `2^n` times (along each dimension), resulting in finer cells
+within the original coarse cell.
```

Comment:
```suggestion
Given a coarse `grid` tesselating the state space, construct a `SubdivisionBasedGrid`
based on the given level array `lvl_array` that should have the same dimension as `grid`. 
The level array has non-negative integer values, with 0 meaning that the 
corresponding cell of the coarse `grid` should not be subdivided any further. 
Value `n > 0` means that the corresponding cell will be subdivided
in total `2^n` times (along each dimension), resulting in finer cells
within the original coarse cell.
```

Reply:
I am not sure about the dimension of the grid. You can correct this.  

Another point: this is the documentation of a type. It should be explicit in the docstring?